.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,51 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+
+name: Defang CLI Package
+
+on:
+  release:
+    types: [created]
+  push:
+    tags:
+      - "v*"
+
+env:
+  DEFANG_CLI_RELEASE_VERSION: ${{ github.ref_name }}
+  NODE_VERSION: "21"
+  NPM_REGISTRY_URL: "https://registry.npmjs.org"
+
+jobs:
+  publish-npm-binaries:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+        working-directory: ./packaging/npm
+
+    steps:
+      - name: Checkout tag
+        uses: actions/checkout@v4
+
+      - name: Install node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          registry-url: ${{ env.NPM_REGISTRY_URL }}
+
+      - name: Publish to NPM
+        run: |
+          #update version
+          npm version ${{ env.DEFANG_CLI_RELEASE_VERSION }}
```

Comment:
Should we commit & push this new version to the repo? If not, we should probably make it clear the "version" in package.json is a placeholder

Reply:
I update the comment here and update the package jsons with a <PLACEHOLDER> value